File: ex1.py
number = int(input("Enter the number of rows: ")) 
print()

# after taking input from user, print upper part 
# printing spaces
for i in range(1, number + 1):
    for j in range(number - i):
        print(" ", end="")
    
    # print stars
    for y in range(1, 2 * i):  
        print("*", end="")
    
    print()

# printing lower part
# printing spaces first
for i in range(number - 1, 0, -1):
    for j in range(number - i): 
        print(" ", end="")
    
    # printing stars
    for y in range(1, 2 * i): 
        print("*", end="")
    
    print()

File: ex1.js
// take user input for the number of rows
let number = parseInt(prompt("Enter the number of rows: "));
let result = "";

// print upper part of the diamond
for (let i = 1; i <= number; i++) {
    // print spaces
    result += " ".repeat(number - i);
    // print stars
    result += "*".repeat(2 * i - 1);
    // move to next line
    result += "\n";
}

// print lower part of the diamond
for (let i = number - 1; i > 0; i--) {
    // print spaces
    result += " ".repeat(number - i);
    // print stars
    result += "*".repeat(2 * i - 1);
    // move to next line
    result += "\n";
}

// output 
console.log(result);


